token_reject_transaction.go

```diff
@@ -0,0 +1,349 @@
+	tx.Transaction.SetNodeAccountIDs(nodeID)
+	return tx
+}
+
+// SetMaxRetry sets the max number of errors before execution will fail.
+func (tx *TokenRejectTransaction) SetMaxRetry(count int) *TokenRejectTransaction {
+	tx.Transaction.SetMaxRetry(count)
+	return tx
+}
+
+// SetMaxBackoff The maximum amount of time to wait between retries.
+// Every retry attempt will increase the wait time exponentially until it reaches this time.
+func (tx *TokenRejectTransaction) SetMaxBackoff(max time.Duration) *TokenRejectTransaction {
+	tx.Transaction.SetMaxBackoff(max)
+	return tx
+}
+
+// SetMinBackoff sets the minimum amount of time to wait between retries.
+func (tx *TokenRejectTransaction) SetMinBackoff(min time.Duration) *TokenRejectTransaction {
+	tx.Transaction.SetMinBackoff(min)
+	return tx
+}
+
+func (tx *TokenRejectTransaction) SetLogLevel(level LogLevel) *TokenRejectTransaction {
+	tx.Transaction.SetLogLevel(level)
+	return tx
+}
+
+func (tx *TokenRejectTransaction) Execute(client *Client) (TransactionResponse, error) {
+	return tx.Transaction.execute(client, tx)
+}
+
+func (tx *TokenRejectTransaction) Schedule() (*ScheduleCreateTransaction, error) {
+	return tx.Transaction.schedule(tx)
+}
+
+// ----------- Overridden functions ----------------
+
+func (tx *TokenRejectTransaction) getName() string {
+	return "TokenRejectTransaction"
+}
+
+func (tx *TokenRejectTransaction) validateNetworkOnIDs(client *Client) error {
+	if client == nil || !client.autoValidateChecksums {
+		return nil
+	}
+
+	if tx.ownerID != nil {
+		if err := tx.ownerID.ValidateChecksum(client); err != nil {
+			return err
+		}
+	}
+
+	for _, tokenID := range tx.tokenIDs {
+		if err := tokenID.ValidateChecksum(client); err != nil {
+			return err
+		}
+	}
+
+	for _, nftID := range tx.nftIDs {
+		if err := nftID.TokenID.ValidateChecksum(client); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (tx *TokenRejectTransaction) build() *services.TransactionBody {
+	body := tx.buildProtoBody()
+
+	return &services.TransactionBody{
+		TransactionFee:           tx.transactionFee,
+		Memo:                     tx.Transaction.memo,
+		TransactionValidDuration: _DurationToProtobuf(tx.GetTransactionValidDuration()),
+		TransactionID:            tx.transactionID._ToProtobuf(),
+		Data: &services.TransactionBody_TokenReject{
+			TokenReject: body,
+		},
+	}
+}
+
+func (tx *TokenRejectTransaction) buildScheduled() (*services.SchedulableTransactionBody, error) {
+	return &services.SchedulableTransactionBody{
+		TransactionFee: tx.transactionFee,
+		Memo:           tx.Transaction.memo,
+		Data: &services.SchedulableTransactionBody_TokenReject{
+			TokenReject: tx.buildProtoBody(),
+		},
+	}, nil
+}
+
+func (tx *TokenRejectTransaction) buildProtoBody() *services.TokenRejectTransactionBody {
+	body := &services.TokenRejectTransactionBody{}
+
+	if tx.ownerID != nil {
+		body.Owner = tx.ownerID._ToProtobuf()
+	}
+
+	if len(tx.tokenIDs) > 0 {
```

Comment:
nit: Isn't the length check is redundant here.

Reply:
removed